examples/step-16/step-16.cc

```diff
@@ -380,7 +380,7 @@ namespace Step16
   {
     MappingQ1<dim> mapping;
 
-    typedef decltype(dof_handler.begin_active()) Iterator;
+    using Iterator = decltype(this->dof_handler.begin_active());
```

Comment:
Ugh. Can't we just hardcode the type here? Surely this is not readable to the average novice user...